From 47dd5846097e229a0c32995d5462dd5f6e00dae2 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 14 Mar 2024 08:37:24 +0100
Subject: [PATCH 0780/1695] dts: stm32: disable IWDG1 and IWDG2 during low
 power for stm32mp15x boards from DH

Update shadow value of otp HW2 to disable IWDG1 and IWDG2 during standby
and stop in stm32mp15xx-dhcom-som.dtsi and in stm32mp15xx-dhcor-som.dtsi

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 280cae24fa592030c057ecb3c1c604d991bd179c dts: stm32: disable IWDG1 and IWDG2 during low power for stm32mp15x boards from DH]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366598
Change-Id: I53e05d0c7f241bb0a0e1bc4e1c65f74b17c55aff
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372978
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi | 10 ++++++++++
 core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi | 10 ++++++++++
 2 files changed, 20 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
index 75969e5d8..57ff02750 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcom-som.dtsi
@@ -8,6 +8,7 @@
 #include "stm32mp15xxaa-pinctrl.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/mfd/st,stpmic1.h>
+#include <dt-bindings/soc/stm32mp-provisioning.h>
 
 / {
 	aliases {
@@ -70,6 +71,15 @@
 		};
 	};
 
+	shadow-prov {
+		compatible = "st,provisioning";
+		hw2_prov {
+			nvmem-cells = <&hw2_otp>;
+			st,shadow-value = <0x000001E0>;
+			st,shadow-lock = <STICKY_LOCK_SR>;
+		};
+	};
+
 	ethernet_vio: vioregulator {
 		compatible = "regulator-fixed";
 		regulator-name = "vio";
diff --git a/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi b/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
index afd549c74..cb377ac3d 100644
--- a/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dhcor-som.dtsi
@@ -10,6 +10,7 @@
 #include "stm32mp15xxac-pinctrl.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/mfd/st,stpmic1.h>
+#include <dt-bindings/soc/stm32mp-provisioning.h>
 
 / {
 	aliases {
@@ -68,6 +69,15 @@
 			no-map;
 		};
 	};
+
+	shadow-prov {
+		compatible = "st,provisioning";
+		hw2_prov {
+			nvmem-cells = <&hw2_otp>;
+			st,shadow-value = <0x000001E0>;
+			st,shadow-lock = <STICKY_LOCK_SR>;
+		};
+	};
 };
 
 &crc1 {
-- 
2.47.2

